### PR TITLE
[14.0] stock_dynamic_routing: fix rule find domain

### PR DESCRIPTION
**Pull request by mt-software-de**
Current behavior:
All machting stock.routing.rule's were found by _find_rule_for_location also they were the parent routing_id is inacitve
With this PR, it will only find rule's where the parent routing_id is active.

### CODE REVIEW

**Comment by mt-software-de** [comment]
@jbaudoux could you please have a look over this PR

File: stock_dynamic_routing/models/stock_routing.py [hunk 1]
```diff
@@ -146,6 +148,25 @@ def _compute_routing_message(self):
...
+                    ("method", "=", "pull"),
+                ],
+                [
+                    ("routing_location_id", "in", push_location_tree.ids),
+                    ("routing_picking_type_id", "=", picking_type.id),
+                    ("method", "=", "push"),
+                ],
+            ]
+        )
+        return expression.AND([[("routing_id.active", "=", True)], domain])
```
Comment (jbaudoux) [inline stock_dynamic_routing/models/stock_routing.py:168]: I think you then have to add auto_join=True on routing_id m2o definition

Comment (mt-software-de) [reply stock_dynamic_routing/models/stock_routing.py:168]: We don't have to:
Without auto_join it creates a sql with `where routing_id in (select id from stock_routing where active=true)`
For example here a full query:

`SELECT "stock_routing_rule".id FROM "stock_routing_rule" WHERE (("stock_routing_rule"."routing_id" in (SELECT "stock_routing".id FROM "stock_routing" WHERE ("stock_routing"."active" = true))) AND (((("stock_routing_rule"."routing_location_id" in (5,2)) AND ("stock_routing_rule"."routing_picking_type_id" = 37)) AND ("stock_routing_rule"."method" = 'pull')) OR ((("stock_routing_rule"."routing_location_id" in (55,54,49,48,1)) AND ("stock_routing_rule"."routing_picking_type_id" = 37)) AND ("stock_routing_rule"."method" = 'push')))) ORDER BY  "stock_routing_rule"."sequence" ,"stock_routing_rule"."id"`

and with the auto_join it creates a sql with `left join stock_routing as sr .... where sr.active=true`
For example here also a full query
`SELECT "stock_routing_rule".id FROM "stock_routing_rule" LEFT JOIN "stock_routing" AS "stock_routing_rule__routing_id" ON ("stock_routing_rule"."routing_id" = "stock_routing_rule__routing_id"."id") WHERE (("stock_routing_rule__routing_id"."active" = true) AND (((("stock_routing_rule"."routing_location_id" in (63,58,57,1)) AND ("stock_routing_rule"."routing_picking_type_id" = 45)) AND ("stock_routing_rule"."method" = 'pull')) OR ((("stock_routing_rule"."routing_location_id" in (4,2)) AND ("stock_routing_rule"."routing_picking_type_id" = 45)) AND ("stock_routing_rule"."method" = 'push')))) ORDER BY  "stock_routing_rule"."sequence" ,"stock_routing_rule"."id"`

**Review by jbaudoux — APPROVED** [review]

**Comment by jbaudoux** [comment]
@sebalix Can you review?

**Review by simahawk — CHANGES_REQUESTED** [review]
LG. Can you please rewrite the commit message to: 

```
stock_dynamic_routing: fix rule find domain

The domain must consider the active flag too.
```

**Comment by mt-software-de** [comment]
@simahawk 
> LG. Can you please rewrite the commit message to:
> 
> ```
> stock_dynamic_routing: fix rule find domain
> 
> The domain must consider the active flag too.
> ```

Done

**Review by simahawk — APPROVED** [review]

**Review by sebalix — APPROVED** [review]

**Pull request merged**